src/test/java/de/tum/in/www1/artemis/hestia/ProgrammingExerciseTaskServiceTest.java

```diff
@@ -0,0 +1,182 @@
+
+        programmingExercise = programmingExerciseRepository.findAll().get(0);
+        updateProblemStatement("""
+                [task][Task 1](testClass[BubbleSort])
+                [task][Task 2](testMethods[Context])
+                """);
+    }
+
+    private void updateProblemStatement(String problemStatement) {
+        programmingExercise.setProblemStatement(problemStatement);
+        programmingExercise = programmingExerciseRepository.save(programmingExercise);
+        programmingExerciseTaskService.updateTasksFromProblemStatement(programmingExercise);
+    }
+
+    @AfterEach
+    public void resetDatabase() {
+        database.resetDatabase();
+    }
+
+    @Test
+    public void testNewExercise() {
+        assertThat(programmingExerciseTaskRepository.findAll()).hasSize(2);
+        var tasks = programmingExerciseTaskRepository.findByExerciseIdWithTestCases(programmingExercise.getId());
+        assertThat(tasks).hasSize(2);
+
+        assertThat(tasks).anyMatch(programmingExerciseTask -> checkTaskEqual(programmingExerciseTask, "Task 1", "testClass[BubbleSort]"));
+        assertThat(tasks).anyMatch(programmingExerciseTask -> checkTaskEqual(programmingExerciseTask, "Task 2", "testMethods[Context]"));
+    }
+
+    @Test
+    public void testAddTask() {
+        var previousTaskIds = programmingExerciseTaskRepository.findByExerciseIdWithTestCases(programmingExercise.getId()).stream().map(ProgrammingExerciseTask::getId)
+                .collect(Collectors.toSet());
+
+        updateProblemStatement("""
+                [task][Task 1](testClass[BubbleSort])
+                [task][Task 2](testMethods[Context])
+                [task][Task 3](testMethods[Policy])
+                """);
+        assertThat(programmingExerciseTaskRepository.findAll()).hasSize(3);
+        var tasks = programmingExerciseTaskRepository.findByExerciseIdWithTestCases(programmingExercise.getId());
+        assertThat(tasks).hasSize(3);
+
+        assertThat(tasks).anyMatch(programmingExerciseTask -> checkTaskEqual(programmingExerciseTask, "Task 1", "testClass[BubbleSort]"));
+        assertThat(tasks).anyMatch(programmingExerciseTask -> checkTaskEqual(programmingExerciseTask, "Task 2", "testMethods[Context]"));
+        assertThat(tasks).anyMatch(programmingExerciseTask -> checkTaskEqual(programmingExerciseTask, "Task 3", "testMethods[Policy]"));
+
+        // Test that the other tasks were not removed and re-added.
+        var newTaskIds = tasks.stream().map(ProgrammingExerciseTask::getId).collect(Collectors.toSet());
+        assertThat(newTaskIds).containsAll(previousTaskIds);
+    }
+
+    @Test
+    public void testRemoveAllTasks() {
+        updateProblemStatement("Empty");
+        assertThat(programmingExerciseTaskRepository.findAll()).isEmpty();
+    }
+
+    @Test
+    public void testReduceToOneTask() {
+        updateProblemStatement("[task][Task 1](testClass[BubbleSort],testMethods[Context],testMethods[Policy])");
```

Comment:
```suggestion
        updateProblemStatement("[task][Task 1](testClass[BubbleSort],testMethods[Context], testMethods[Policy])");
```
=> Ensures the trimming